scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -104,16 +104,15 @@ class AsyncFlowDefRunner extends Writer { self =>
    * SourcePipes or IterablePipes. These are for both toIterableExecution and forceToDiskExecution
    */
   private case class State(
-    filesToCleanup: Map[Mode, FilesToCleanUp],
-    initToOpt: HMap[TypedPipe, TypedPipe],
+    filesToCleanup: FilesToCleanUp,
+    initToOpt: HMap[StateKey, TypedPipe],
```

Comment:
@fwbrasil note this was a potential bug: we were ignoring the Config in the mapping from original to optimized pipe, but Config can control the optimizations, so this would not be safe in general. We need to include the Config as part of the key as we do in `forcedPipes`.